### ISSUE TITLE
`areIntervalsOverlapping` returns false for overlapping interval

### ISSUE THREAD

**Issue by skbogner** [date-fns/date-fns#3614]
There is an issue with the [areIntervalsOverlapping function](https://date-fns.org/v3.0.0/docs/areIntervalsOverlapping)

  ## The problem

  ### 💻 Code demonstrating the problem

https://stackblitz.com/edit/date-fns-playground-c8oyu9?file=index.ts

  ```js
 import { areIntervalsOverlapping } from 'date-fns';

const left = {
  start: '1970-01-01T02:00:00.000Z',
  end: '1970-01-01T03:00:00.000Z',
};
const right = {
  start: '1969-12-31T23:30:00.000Z',
  end: '1970-01-01T02:30:00.000Z',
};

console.log(areIntervalsOverlapping(left, right, { inclusive: true }));
// => false
  ```

  ### 🙁 Actual behavior

`areIntervalsOverlapping` returns `false`

  ### 🙂 Expected behavior

`areIntervalsOverlapping` returns `true`

  ## Debug information

  - date-fns version: 3.0.0
  - Browser/Node.js version: Multiple/Any
  - Your timezone: Europe/Copenhagen
  - Your current time: 10:00-11:00

**Comment by pekala** [comment]
For additional context, seems related to crossing 1970:

```ts
areIntervalsOverlapping(
    {start: new Date('2024-01-01T02:00:00.000Z'), end: new Date('2024-01-01T03:00:00.000Z')}, 
    {start: new Date('2023-12-31T23:30:00.000Z'),  end: new Date('2024-01-01T02:30:00.000Z')}
)
// true

areIntervalsOverlapping(
    {start: new Date('1970-01-01T02:00:00.000Z'), end: new Date('1970-01-01T03:00:00.000Z')}, 
    {start: new Date('1969-12-31T23:30:00.000Z'),  end: new Date('1970-01-01T02:30:00.000Z')}
)
// false

areIntervalsOverlapping(
    {start: new Date('1971-01-01T02:00:00.000Z'), end: new Date('1971-01-01T03:00:00.000Z')}, 
    {start: new Date('1970-12-31T23:30:00.000Z'),  end: new Date('1971-01-01T02:30:00.000Z')}
)
// true

```

**Comment by kossnocorp** [comment]
Fixed and shipped with `date-fns@3.0.6`
